### PR TITLE
add variable order to tree structure

### PR DESCRIPTION
**Pull request by dehann**
addresses #495

### CODE REVIEW

File: src/JunctionTreeTypes.jl [hunk 1]
```diff
@@ -14,14 +14,16 @@ mutable struct BayesTree
   btid::Int
   cliques::Dict{Int,Graphs.ExVertex}
   frontals::Dict{Symbol,Int}
+  variableOrder::Vector{Symbol}
```
Comment (tonioteran) [inline src/JunctionTreeTypes.jl]: I see how this is adding the variableOrder member variable, but it still fails the check with an error saying that BayesTree does not have that field.

**Comment by tonioteran** [comment]
Seems good to me, but for some reason I get the following error while doing a quick check---which I added as a test case:
```julia
    fg = loadCanonicalFG_Kaess()
    # Choose specific variable ordering and perform check.
    vo = [:l1, :l2, :x1, :x2, :x3]
    tree = buildTreeFromOrdering!(fg, vo)
```

accompanied by the following error:

```julia
ERROR: type BayesTree has no field variableOrdering
Stacktrace:
 [1] setproperty!(::BayesTree, ::Symbol, ::Array{Symbol,1}) at ./Base.jl:21
 [2] #buildTreeFromOrdering!#477(::Bool, ::Int64, ::Int64, ::typeof(buildTreeFromOrdering!), ::DistributedFactorGraphs.GraphsDFG{SolverParams}, ::Array{Symbol,1}) at /home/tonio/.julia/dev/IncrementalInference/src/JunctionTree.jl:380
 [3] buildTreeFromOrdering!(::DistributedFactorGraphs.GraphsDFG{SolverParams}, ::Array{Symbol,1}) at /home/tonio/.julia/dev/IncrementalInference/src/JunctionTree.jl:373
 [4] top-level scope at none:0
```

I'm not sure why it's not wanting to detect the new variable member, or if I'm missing something really silly.

**Comment by dehann** [comment]
Hi @tonioteran , dang sorry i missed spelling and didn’t see this till now.  in the one place i wrote variableOrder (actual field name), but elsewhere i wrote variableOrdering.  silly mistake and lots of time, sorry about that!

**Comment by dehann** [comment]
probably best to go with ‘variableOrder’ everywhere i think.

File: src/JunctionTreeTypes.jl (hunk 1 above)
Comment (dehann) [reply src/JunctionTreeTypes.jl]: i committed a fix

**Comment by tonioteran** [comment]
Can I go ahead and delete this branch?

**Pull request merged**